transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
The documentation never mentions that an 'initial' state is created if initial=None if I am not mistaken. Maybe we should just remove that behaviour completely. What do you think @tyarkoni and @wtgee?